homeassistant/components/vicare/utils.py

```diff
@@ -1,13 +1,28 @@
 """ViCare helpers functions."""
 import logging
 
+from PyViCare.PyViCareDevice import Device as PyViCareDevice
+from PyViCare.PyViCareDeviceConfig import PyViCareDeviceConfig
 from PyViCare.PyViCareUtils import PyViCareNotSupportedFeatureError
 
+from homeassistant.config_entries import ConfigEntry
+
 from . import ViCareRequiredKeysMixin
+from .const import CONF_HEATING_TYPE, HEATING_TYPE_TO_CREATOR_METHOD, HeatingType
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def get_device(
+    entry: ConfigEntry, device_config: PyViCareDeviceConfig
+) -> PyViCareDevice:
+    """Get device for device config."""
+    return getattr(
+        device_config,
+        HEATING_TYPE_TO_CREATOR_METHOD[HeatingType(entry.data[CONF_HEATING_TYPE])],
```

Comment:
This uses the device type provided in the configuration. In my setup it's fine as I have two gas boilers. Others could have issues when multiple devices are present and the type can not be determined via the `auto` type.

I have no idea how to solve this and ideas are welcome.

Reply:
Maybe this is acceptable as currently there is support for one device only?

Reply:
It's fine for me. Are you mentioning which device types are supported in the docs or elsewhere?

Reply:
Yes, in the docs there is a list of supported types, but no detailed list.

https://github.com/home-assistant/home-assistant.io/blob/c2596f56fd802a03d3a6932fac1c7bd91d493fe2/source/_integrations/vicare.markdown?plain=1#L47